Remove non-standard legacy `drawImageFromRect` (CanvasDrawImage)

Ahmad-S792 opened the pull request:
#### 8f52c3b201776ea6135d3a0a793e876f93940c62
<pre>
Remove non-standard legacy `drawImageFromRect` (CanvasDrawImage)

<a href="https://bugs.webkit.org/show_bug.cgi?id=284878">https://bugs.webkit.org/show_bug.cgi?id=284878</a>
<a href="https://rdar.apple.com/141681635">rdar://141681635</a>

Reviewed by Kimmo Kinnunen.

This patch is to align WebKit with Gecko / Firefox and Blink / Chromium.

We have our current implementation based on standards (drawImage) so this patch
aims to get rid of legacy non-standard `drawImageFromRect`.

Blink removed this as well in 2014 in below commit:

Commit: <a href="https://github.com/chromium/chromium/commit/ce07cefc396fdf4016e731b4e75592586785c6b0">https://github.com/chromium/chromium/commit/ce07cefc396fdf4016e731b4e75592586785c6b0</a>

From MDN data, Safari / WebKit has supported standard alternatives since Safari 2,
so this is about time to try to get rid of non-standard legacy `drawImageFromRect`.

* Source/WebCore/html/canvas/CanvasRenderingContext2D.idl:
* Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp:
(WebCore::CanvasRenderingContext2DBase::drawImageFromRect): Deleted.
* Source/WebCore/html/canvas/CanvasRenderingContext2DBase.h:
* Source/WebInspectorUI/UserInterface/Models/NativeFunctionParameters.js:
* Source/WebInspectorUI/UserInterface/Models/RecordingAction.js:
(WI.RecordingAction.prototype.getImageParameters):
* Source/WebInspectorUI/UserInterface/Views/RecordingActionTreeElement.js:
(WI.RecordingActionTreeElement._classNameForAction):
* LayoutTests/fast/canvas/canvas-overloads-drawImageFromRect-expected.txt: Removed.
* LayoutTests/fast/canvas/canvas-overloads-drawImageFromRect.html: Removed.
* LayoutTests/fast/canvas/drawImageFromRect_withToDataURLAsSource-expected.txt: Removed.
* LayoutTests/fast/canvas/drawImageFromRect_withToDataURLAsSource.html: Removed.
* LayoutTests/fast/canvas/image-object-in-canvas-expected.txt:
* LayoutTests/fast/canvas/image-object-in-canvas.html:
* LayoutTests/inspector/canvas/resources/recording-2d.js:
* LayoutTests/inspector/canvas/recording-2d-full-expected.txt:

Canonical link: <a href="https://commits.webkit.org/297846@main">https://commits.webkit.org/297846@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4d62a538156b408fd01eb83a4699182811ab9a33

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/113099 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/32834 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/23312 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/119306 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/63697 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Compiled WebKit") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/115061 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/33486 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/41397 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/86080 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/37285 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/116046 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/26741 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/101736 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/66395 "Found 1 new API test failure: /WPE/TestSSL:/webkit/WebKitWebView/ephemeral-tls-errors (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/26017 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/19867 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/63061 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/96128 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/19940 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/122524 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/40177 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/29967 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/94927 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/40561 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/97955 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/94668 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24160 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/39816 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17621 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/36302 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/40063 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/45562 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/39704 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/43037 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/41441 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

kkinnunen-apple commented on LayoutTests/inspector/canvas/recording-2d-full-expected.txt:
```diff
@@ -305,19 +305,6 @@ frames:
         3: (anonymous function)
         4: executeFrameFunction
   15: (duration)
-    0: drawImageFromRect([object HTMLImageElement], 1, 2, 3, 4, 5, 6, 7, 8, "")
```
maybe you need to remove the calls in recording-2d.js ?

Ahmad-S792 replied:
Done. :-)

kkinnunen-apple reviewed (approved):
Approved with a request to git grep one more time and remove all references if removing them makes sense.

Ahmad-S792 committed:
Remove non-standard legacy `drawImageFromRect` (CanvasDrawImage)

https://bugs.webkit.org/show_bug.cgi?id=284878
rdar://141681635

Reviewed by Kimmo Kinnunen.

This patch is to align WebKit with Gecko / Firefox and Blink / Chromium.

We have our current implementation based on standards (drawImage) so this patch
aims to get rid of legacy non-standard `drawImageFromRect`.

Blink removed this as well in 2014 in below commit:

Commit: https://github.com/chromium/chromium/commit/ce07cefc396fdf4016e731b4e75592586785c6b0

From MDN data, Safari / WebKit has supported standard alternatives since Safari 2,
so this is about time to try to get rid of non-standard legacy `drawImageFromRect`.

* Source/WebCore/html/canvas/CanvasRenderingContext2D.idl:
* Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp:
(WebCore::CanvasRenderingContext2DBase::drawImageFromRect): Deleted.
* Source/WebCore/html/canvas/CanvasRenderingContext2DBase.h:
* Source/WebInspectorUI/UserInterface/Models/NativeFunctionParameters.js:
* Source/WebInspectorUI/UserInterface/Models/RecordingAction.js:
(WI.RecordingAction.prototype.getImageParameters):
* Source/WebInspectorUI/UserInterface/Views/RecordingActionTreeElement.js:
(WI.RecordingActionTreeElement._classNameForAction):
* LayoutTests/fast/canvas/canvas-overloads-drawImageFromRect-expected.txt: Removed.
* LayoutTests/fast/canvas/canvas-overloads-drawImageFromRect.html: Removed.
* LayoutTests/fast/canvas/drawImageFromRect_withToDataURLAsSource-expected.txt: Removed.
* LayoutTests/fast/canvas/drawImageFromRect_withToDataURLAsSource.html: Removed.
* LayoutTests/fast/canvas/image-object-in-canvas-expected.txt:
* LayoutTests/fast/canvas/image-object-in-canvas.html:
* LayoutTests/inspector/canvas/resources/recording-2d.js:
* LayoutTests/inspector/canvas/recording-2d-full-expected.txt:

Canonical link: https://commits.webkit.org/297846@main
